bindings/rust/src/lib.rs

```diff
@@ -40,14 +40,66 @@ mod sys;
 use std::ffi::CString;
 use std::ptr::NonNull;
 
+/// A safe container for handling the low-level FizzyError struct.
+struct CErrorContainer(Box<sys::FizzyError>);
+
+impl CErrorContainer {
+    /// Create a safe, boxed, and zero initialised container.
+    fn new() -> Self {
+        CErrorContainer {
+            0: Box::new(sys::FizzyError {
+                code: 0,
+                message: [0i8; 256],
+            }),
+        }
+    }
+
+    /// Return a pointer passable to low-level functions (validate, parse, instantiate).
+    ///
+    /// # Safety
+    /// The returned pointer is still onwed by this struct, and thus not valid after this struct goes out of scope.
+    unsafe fn as_mut_ptr(&mut self) -> *mut sys::FizzyError {
+        &mut *self.0
+    }
+
+    /// Return the underlying error code.
+    // TODO: represent the errors as proper Rust enums
+    fn code(&self) -> u32 {
+        self.0.code
+    }
+
+    /// Return an owned String copy of the underlying message.
+    fn message(&self) -> String {
+        unsafe {
+            std::ffi::CStr::from_ptr(self.0.message.as_ptr())
+                .to_string_lossy()
+                .into_owned()
+        }
+    }
+}
+
+impl std::fmt::Display for CErrorContainer {
```

Comment:
This is not currently used, but it enables things like `println!("{:?}", err)` or `err.to_string()`. Probably should remove.

Reply:
Sounds useful, can there be a test for it?

Reply:
Added.